File: sdks/ruby/lib/novabilling/types/pagination_meta.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class PaginationMeta < Internal::Types::Model
      field :total, -> { Integer }, optional: false, nullable: false
      field :page, -> { Integer }, optional: false, nullable: false
      field :limit, -> { Integer }, optional: false, nullable: false
      field :total_pages, -> { Integer }, optional: false, nullable: false, api_name: "totalPages"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/PaginationMeta.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record PaginationMeta : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("total")]
    public required double Total { get; set; }

    [JsonPropertyName("page")]
    public required double Page { get; set; }

    [JsonPropertyName("limit")]
    public required double Limit { get; set; }

    [JsonPropertyName("totalPages")]
    public required double TotalPages { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
